### PR TITLE
fix names of model-entity joint setters

### PR DESCRIPTION
**Pull request by sethalves**
- fix names of ModelEntity joint setters to match what they actually do
- add missing getters and setters for relative and absolute model-entity joint poses

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 844cc03dcbbc7a8764f20cabbab13f4ce9736734 available at: https://gist.github.com/7f1b1a5ac8bd07b55fef53d58bde623d<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8984-844cc03.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8984-844cc03.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 339d6a55cfec28d12dfd1fbbd75fb298455e2ba2 available at: https://gist.github.com/7f1b1a5ac8bd07b55fef53d58bde623d<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8984-339d6a5.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8984-339d6a5.dmg

File: libraries/entities-renderer/src/RenderableModelEntityItem.cpp [hunk 1]
```diff
@@ -1018,32 +1018,66 @@ glm::vec3 RenderableModelEntityItem::getAbsoluteJointTranslationInObjectFrame(in
 }
 
 bool RenderableModelEntityItem::setAbsoluteJointRotationInObjectFrame(int index, const glm::quat& rotation) {
+    // TODO -- write this
+    assert(false);
+    abort();
```
Comment (ZappoMan) [inline libraries/entities-renderer/src/RenderableModelEntityItem.cpp]: that escalated quickly! - do we want this in here?

Comment (sethalves) [inline libraries/entities-renderer/src/RenderableModelEntityItem.cpp]: we don't, but I haven't figured out how to write this, yet.

Comment (Nex-Pro) [inline libraries/entities-renderer/src/RenderableModelEntityItem.cpp]: Is the title of this correct ? Did you add getters and setters ?

**Comment by hifi-gustavo** [comment]
Build console output for commit: e41fa94958279ef74098afe78ffe5e8197f5e13c available at: https://gist.github.com/7f1b1a5ac8bd07b55fef53d58bde623d<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8984-e41fa94.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8984-e41fa94.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 689fdfb35ff1242c6891bdc0b387b0755519c024 available at: https://gist.github.com/7f1b1a5ac8bd07b55fef53d58bde623d<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8984-689fdfb.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8984-689fdfb.dmg

File: libraries/animation/src/Rig.cpp [hunk 2]
```diff
@@ -308,6 +308,13 @@ void Rig::clearIKJointLimitHistory() {
     }
 }
 
+int Rig::getJointParentIndex(int childIndex) {
```
Comment (hyperlogic) [inline libraries/animation/src/Rig.cpp]: make this method const, please

File: libraries/entities/src/ModelEntityItem.h [hunk 3]
```diff
@@ -117,8 +117,8 @@ class ModelEntityItem : public EntityItem {
 
     virtual bool shouldBePhysical() const override;
 
-    virtual glm::vec3 getJointPosition(int jointIndex) const { return glm::vec3(); }
-    virtual glm::quat getJointRotation(int jointIndex) const { return glm::quat(); }
+    // virtual glm::vec3 getJointPosition(int jointIndex) const { return glm::vec3(); }
+    // virtual glm::quat getJointRotation(int jointIndex) const { return glm::quat(); }
```
Comment (hyperlogic) [inline libraries/entities/src/ModelEntityItem.h]: just delete these virtual methods if no one is overriding them.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 51f6b5af78222d8d2632da7f4fb00370a09353ea available at: https://gist.github.com/7f1b1a5ac8bd07b55fef53d58bde623d<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8984-51f6b5a.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8984-51f6b5a.dmg

**Comment by sethalves** [comment]
to test:
  var clair = Entities.addEntity({
      modelURL: "https://hifi-public.s3.amazonaws.com/cozza13/apartment/claire/Claire.fbx",
      name: "Claire",
      position: Vec3.sum(MyAvatar.position, {x: 3, y: 0, z: 0}),
      type: "Model"
  });

  var headIndex = Entities.getJointIndex(clair, "Head");

  var originalHeadRotation = Entities.getAbsoluteJointRotationInObjectFrame(clair, headIndex);
  Entities.setAbsoluteJointRotationInObjectFrame(clair, headIndex, Quat.fromVec3Degrees({x: 0, y: 45, z: 0}));
  ... she looks left
  Entities.setAbsoluteJointRotationInObjectFrame(clair, headIndex, originalHeadRotation);
  ... she looks straight ahead

  var originalHeadTranslation = Entities.getAbsoluteJointTranslationInObjectFrame(clair, headIndex);
  Entities.setAbsoluteJointTranslationInObjectFrame(clair, headIndex, {x: 0, y: 1, z: 0});
  ... head up high
  Entities.setAbsoluteJointTranslationInObjectFrame(clair, headIndex, originalHeadTranslation);
  ... head back in place

  var originalLocalHeadRotation = Entities.getLocalJointRotation(clair, headIndex);
  Entities.setLocalJointRotation(clair, headIndex, Quat.fromVec3Degrees({x: 0, y: 45, z: 0}));
  ... she looks left
  Entities.setLocalJointRotation(clair, headIndex, originalLocalHeadRotation);
  ... she looks straight ahead

  var originalLocalHeadTranslation = Entities.getLocalJointTranslation(clair, headIndex);
  Entities.setLocalJointTranslation(clair, headIndex, {x: 0, y: 1, z: 0});
  ... she ducks
  Entities.setLocalJointTranslation(clair, headIndex, originalLocalHeadTranslation);
  ... head back in place

**Comment by OmegaHeron** [comment]
Testing Windows 10

QA Pass
All actions via script commands in console did what test described.

**Pull request merged**